src/higher-level/lsm-hl.c

```diff
@@ -423,7 +424,10 @@ float unit_perim = 0;
         if (phl->lsm.oavg) lsm.avg_[f][p] = mx;
         if (phl->lsm.ogeo) lsm.geo_[f][p] = exp(logSum / (float)logCounter);
         if (phl->lsm.omax) lsm.max_[f][p] = maxVal;
-        if (phl->lsm.oare) lsm.are_[f][p] = validDataPixels;
+        if (phl->lsm.oare) {
+            if (validDataPixels <= SHRT_MAX) lsm.are_[f][p] = validDataPixels;
+            else validDataPixels = SHRT_MAX;
```

Comment:
- please use curly brackets for the if else construct
- 2nd line should be `sm.are_[f][p] = SHRT_MAX;`